src/FrameBuffer.cpp

```diff
@@ -418,6 +423,42 @@ CachedTexture * FrameBuffer::_getSubTexture(u32 _t)
 	return m_pSubTexture;
 }
 
+CachedTexture * FrameBuffer::_copyFrameBufferTexture()
+{
+	if (m_copied)
+		return m_pFrameBufferCopyTexture;
+
+	if (m_pFrameBufferCopyTexture == nullptr) {
+		m_pFrameBufferCopyTexture = textureCache().addFrameBufferTexture(config.video.multisampling != 0);
+		_initTexture(m_width, VI_GetMaxBufferHeight(m_width), m_pTexture->format, m_pTexture->size, m_pFrameBufferCopyTexture);
+		_setAndAttachTexture(m_copyFBO, m_pFrameBufferCopyTexture, 0, config.video.multisampling != 0);
+		if (config.video.multisampling != 0)
+			m_pFrameBufferCopyTexture->frameBufferTexture = CachedTexture::fbMultiSample;
+	}
+
+	Context::BlitFramebuffersParams blitParams;
+	blitParams.readBuffer = m_FBO;
+	blitParams.drawBuffer = m_copyFBO;
+	blitParams.srcX0 = 0;
+	blitParams.srcY0 = 0;
+	blitParams.srcX1 = m_pTexture->realWidth;
+	blitParams.srcY1 = m_pTexture->realHeight;
+	blitParams.dstX0 = 0;
+	blitParams.dstY0 = 0;
+	blitParams.dstX1 = m_pTexture->realWidth;
+	blitParams.dstY1 = m_pTexture->realHeight;
+	blitParams.mask = blitMask::COLOR_BUFFER;
+	blitParams.filter = textureParameters::FILTER_NEAREST;
+
+	gfxContext.blitFramebuffers(blitParams);
```

Comment:
What if GPU does not support blitFramebuffers? GL ES 2.0 for example.
May be use GraphicsDrawer::blitOrCopyTexturedRect ?

Reply:
I don't think copyTexturedRect will work to solve this problem. In order to use that, you still need to sample from m_pTexture to do the copy, which is the whole problem this is trying to solve. glBlitFramebuffer is the only way to do it. I should disable this solution if glBlitFramebuffer isn't supported though